src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -173,6 +176,38 @@ public void Initialize(StorageEnvironmentOptions options)
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public long? GetChangeVectorForCurrentDb(DocumentsOperationContext context)
+        {
+            return GetChangeVectorFor(context,Environment.DbId);
+        }
+
+        public long? GetChangeVectorFor(DocumentsOperationContext context, Guid dbId)
+        {			
+            var tree = context.Transaction.InnerTransaction.CreateTree("ChangeVector");
+
+            //not sure if possible to reduce allocations here
+            var readResult = tree.Read(new Slice(dbId.ToByteArray()));
+
+            return readResult?.Reader.ReadBigEndianInt64();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void SetChangeVectorForCurrentDb(DocumentsOperationContext context, long val)
+        {
+            SetChangeVectorFor(context, Environment.DbId, val);
+        }
+
+        public void SetChangeVectorFor(DocumentsOperationContext context, Guid dbId, long val)
+        {
+            //not sure if possible to reduce allocations here
```

Comment:
You can pass the data directly as a pointer, since this is on the stack.
The allocation here is actually with the `ToByteArray()`